src/core/core/logging.cpp

```diff
@@ -39,13 +39,29 @@ DSN_DEFINE_string("core",
 
 DSN_DEFINE_bool("core", logging_flush_on_exit, true, "flush log when exit system");
 
+namespace dsn {
+std::function<std::string()> log_prefixed_message_func = []() {
+    static thread_local std::string prefixed_message(' ', 23);
+    int tid = dsn::utils::get_current_tid();
+    sprintf(const_cast<char *>(prefixed_message.c_str()), "unknown.io-thrd.%05d: ", tid);
```

Comment:
Thoungh prefixed_message is a thread_local type, but these two lines will execute for every calls, you can use a std::call_once here.

Reply:
good idea